esp/esdllib/esdl_def.cpp

```diff
@@ -959,14 +959,87 @@ void EsdlDefStruct::load(EsdlDefinition *esdl, XmlPullParser *xpp, StartTag &str
     }
 }
 
+static SecAccessFlags translateAuthLevel(const char* flag)
+{
+    if (!flag || !*flag)
+        return SecAccess_Full;
+    if (!stricmp(flag,"None") || !stricmp(flag,"Deferred"))
+        return SecAccess_None;
+    if (!stricmp(flag,"Access"))
+        return SecAccess_Access;
+    if (!stricmp(flag,"Read"))
+        return SecAccess_Read;
+    if (!stricmp(flag,"Write"))
+        return SecAccess_Write;
+    if (!stricmp(flag,"Full"))
+        return SecAccess_Full;
+
+    DBGLOG("Unknown access level: %s", flag);
+    return SecAccess_Full;
+}
+
+static void parseAccessList(const char * rawServiceAccessList, MapStringTo<SecAccessFlags> & accessmap, const char * name)
+{
```

Comment:
I think there are a couple of ways you could make this routine easier to read and less prone to possible issues.

Instead of having a single loop and a state variable "nameComplete", you could
1. have a primary loop and  
   A. within that loop an initial loop to read the name until: , : or "   (or just call a separate function "readAccessName()")
   B. then if ':' read the value (or "readAccessValue()")
  C. then process pair
  D. loop for next pair.
2. Parse the pairs first into a StringArray.  StringArray::appendList(rawServiceAccessList, ",");
    -process each pair, separating name and value, etc.
    - will have to deal with the ending "